File: Problem_3.py
def isprime(num):
    prime = True
    for i in range(2,int(num ** 0.5)):
        if num % i == 0:
            prime = False
    
    return prime

number = 600851475143
max_factor = int(number ** 0.5)
for i in range(max_factor, 0, -1):
    if number % i == 0:
        if isprime(i) == True:
            max_prime_factor = i
            break

print("The greatest prime factor of 600851475143 is ", max_prime_factor)

File: Problem_3.cpp
/*
 Problem_3.cpp
 Created by Harrison Hanks on 7/29/16
 
 What is the largest prime factor of the number 600851475143?
*/

#include <iostream>
#include <cmath>
using namespace std;

bool isprime(int);

int main()
{
    long long unsigned int number = 600851475143;
    //long long unsigned int max_factor = sqrt(number);
    long long unsigned int max_factor = number / 2;
    long long unsigned int i;
    int max_prime_factor;
    
    for (i = max_factor; i > 1; i--)
    {
        if (number % i == 0)
        {
            if (isprime(i) == true)
            {
                max_prime_factor = i;
                break;
            }
        }
    }
    
    cout << "The greatest prime factor of " << number << " is " << max_prime_factor << endl;
}

// Very primitive primality test
bool isprime(int num)
{
    int max = sqrt(num);
    
    for (int divisor = 2; divisor <= max; divisor++)
    {
        if (num % divisor == 0)
        {
            return false;
        }
    }
    
    return true;
}